src/storage/gpu_swap_history.cc

```diff
@@ -46,21 +80,66 @@ void MemHistory::PutRecord(handle_id_t handle_id, int device,
   record_idx[device]++;
 }
 
-// optimal algorithm: assume iterations remain the same; choose the handle
-// whose next reference is furthest in the future as victim.
-handle_id_t MemHistory::DecideVictim(std::unordered_set<handle_id_t> handles, int device) {
-  std::lock_guard<std::mutex> lock(mutex_[device]);
+handle_id_t MemHistory::LRU(std::unordered_set<handle_id_t> handles, int device) {
```

Comment:
Add a comment before the function to simply describe the algorithm (even if it is LRU).